lib/cisco_node_utils/cmd_ref/vlan.yaml

```diff
@@ -42,18 +42,61 @@ name:
   get_command: "show vlan brief"
   get_value: '/^%d\s+(\S+)\s/'
   set_context: ["vlan %d"]
-  set_value: "%s name %s"
+  set_value: "%s name %s ; end"
+
+private_vlan_association_add_vlans:
```

Comment:
Capturing our earlier discussion here:

 
- We should only have one property that can be used by the manifest to describe the final state of associated private vlans for a given vlan.
- The setter for this property should take a list, not a hash.
Do not support the ‘add/remove’ key word.

**Example Manifest:**
``` 
cisco_private_vlan { 11:
  …
  private_vlan_association => “5,6,13-15”,
}
```
In the case above, 5,6,13-15 would be associated with vlan 11 and any other private vlans associated with vlan 11 should be purged.
 
The private_vlan_association getter and setter should handle the following:
- Getter: Query existing private vlans association for a given vlan.
   - Example: Call private_vlan_association returns list “5,12,14,18"
- Setter: Add a list of associated private vlans for a given vlan. 
  - Example: private_vlan_association = “5,6,10,11” 
  - This would add, “5,6,10,11” and remove any other associated private vlans that are not specified in the manifest list but exist on the device.
  - Here is a method `get_reset_range` that you should be able to leverage or use to determine the delta of things that need to be added and removed.
https://github.com/cisco/cisco-network-node-utils/blob/e8fdb83fc2677514877e08df33801408f7616b30/lib/cisco_node_utils/stp_global.rb#L58
- Setter: Remove ALL associated private vlans for a given vlan.
  - Example: private_vlan_association = “” # Set it to an empty list”
  - Puppet will first call the getter to return the list of existing associated private vlans.  This list should be compared to the manifest with the following outcomes.
    - No action if manifest values and getter values match. (idempotent)
    - Associate private vlans specified in the manifest that are not present in the list returned by the getter and purge any additional vlans on the switch.
    - Remove all existing associated vlans when the manifest specifies an empty list.
 
Here is that the yaml entry should look like.

```
private_vlan_association:
  _exclude: [N8k]
  kind: string
  multiple: true
  get_command: "show vlan private-vlan"
  get_value: '/^<id>\s+(\d+)/'
  set_context: ['vlan <vlan>']
  # Need end to push the config Without the config wont take effect
  set_value: “<state> private-vlan association <vlans> ; end"
  default_value: ""
```